### ISSUE TITLE
Caja device unmount symbol is not working

### ISSUE THREAD

**Issue by projectkk2glider** [mate-desktop/caja#904]
#### Expected behaviour
Device should unmount.

#### Actual behaviour
Nothing happens.

#### Steps to reproduce the behaviour
Mount removable device. Open Caja and then click on `unmount` symbol in the Places -> Devices list. Device is not unmounted. Here is a screenshot. The mouse cursor is on the unmount symbol that is not working.

![issue_caja](https://user-images.githubusercontent.com/5950438/34962652-f07074e8-fa44-11e7-89b0-ec4ca109672d.png)

If I right click on the device and then click on `unmount` menu option, then the device is unmounted - works as expected.

#### MATE general version
1.18

#### Package version
```
dpkg -l | grep caja
ii  caja                                        1.18.5-1+sonya                               amd64        file manager for the MATE desktop
ii  caja-common                                 1.18.5-1+sonya                               all          file manager for the MATE desktop (common files)
ii  caja-extensions-common                      1.18.2-1+sonya                               all          Caja extensions (common files)
ii  caja-folder-color-switcher                  1.1.7                                        all          Change a folder color
ii  caja-gksu                                   1.18.2-1+sonya                               amd64        privilege granting extension for Caja using gksu
ii  caja-open-terminal                          1.18.2-1+sonya                               amd64        Caja plugin for opening terminals in arbitrary local paths
ii  caja-sendto                                 1.18.2-1+sonya                               amd64        integrates email clients and Pidgin into the Caja file manager
ii  caja-wallpaper                              1.18.2-1+sonya                               amd64        Allows one to quickly set a wallpaper from folders
ii  gir1.2-caja                                 1.18.5-1+sonya                               amd64        GObject introspection data for Caja
ii  libcaja-extension1:amd64                    1.18.5-1+sonya                               amd64        libraries for Caja components
ii  python-caja                                 1.18.1-1+sonya                               amd64        Python binding for Caja components
ii  python-caja-common                          1.18.1-1+sonya                               all          Python binding for Caja components (common files)
```

#### Linux Distribution
```
cat /etc/issue
Linux Mint 18.3 Sylvia \n \l

uname -a
Linux xxxx 4.13.0-26-generic #29~16.04.2-Ubuntu SMP Tue Jan 9 22:00:44 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

#### Link to downstream report of your Distribution
https://bugs.launchpad.net/linuxmint/+bug/1743470

**Comment by lukefromdc** [comment]
Try resizing the sidebar narrower, the "eject" icon should then work. If so, you have run into
https://github.com/mate-desktop/caja/issues/880
which was fixed with
https://github.com/mate-desktop/caja/commit/c6b81f64c4c4212e414c04e584380774d6ac83ca
but is not included in Caja 1.18.5

**Comment by projectkk2glider** [comment]
You might be right. When I resize the sidebar to the narrowest possible width, I am able to unmount by clicking just slightly left of the eject (unmount) icon.

I then tried with various sidebar widths and realized that the eject area and eject icon are not aligned! For example, when I make the sidebar extra wide, the location that actualy unmounts is where the mouse cursor lies:

![issue_caja2](https://user-images.githubusercontent.com/5950438/34974140-ac6cb828-fa8b-11e7-8246-30281c1cc544.png)

So basically the eject is working, but I must click somewhere left from the eject icon. How can I test the above fix?

**Comment by lukefromdc** [comment]
https://github.com/mate-desktop/caja/commit/c6b81f64c4c4212e414c04e584380774d6ac83ca is in 1.19.3 and in git master. If you can build Caja locally, give either one a try

**Comment by projectkk2glider** [comment]
Built current master (9b12785db1907b279fb7d6d80fd1d9fd4bb04bfb) on a fresh Mint 18.3 MATE installation (that had this problem with the default caja version) and the **problem is gone**!

So looks like the problem is fixed. Will you release the fix for 1.18.x branch so we can have it in the Mint distribution?


----
Maybe this will help somebody in the future. To **build and test caja** I used fresh virtual machine with Mint 18.3 MATE and:
```
apt install libxml2-dev libxslt1-dev libglib2.0-dev libidl-dev \
     libdbus-1-dev libdbus-glib-1-dev libpolkit-backend-1-dev flex libpopt-dev \
     bison libbz2-dev libgcrypt11-dev libcanberra-dev libgail-dev libgtk2.0-dev \
     libart-2.0-dev libglade2-dev libtasn1-3-bin libxklavier-dev libsoup2.4-dev \
    icon-naming-utils libunique-dev libcanberra-gtk-dev libwnck-dev \
    librsvg2-dev libpolkit-agent-1-dev gobject-introspection \
    libgirepository1.0-dev libupower-glib-dev intltool xsltproc \
    libtasn1-doc libtool gtk-doc-tools libgamin-dev \
    librarian-dev libdconf-dev libsecret-1-dev libgnome-keyring-dev \
    yelp-tools libnotify-dev

apt install libgtk-3-dev libgail-3-dev libmate-desktop-dev

git clone https://github.com/mate-desktop/caja.git
cd caja/
./autogen.sh --prefix=/usr
make
sudo make install
caja -q && caja
```

**Comment by monsta** [comment]
Ok, so it was the same problem, closing it then.
We'll release new 1.18.x later, can't promise when though. We're currently busy with preparing 1.20 release.

**Comment by projectkk2glider** [comment]
OK, thanks for quick a response to my problem.
